docs/go/core/bzlmod.md

```diff
@@ -161,6 +160,37 @@ go_deps.module(
 )
 ```
 
+#### Depending on tools
+
+If you need to depend on Go modules that are only used as tools, you can use the [`tools.go` technique](https://github.com/golang/go/issues/25922#issuecomment-1038394599):
+
+1. In a new subdirectory of your repository, create a `tools.go` file that imports the tools' main packages:
+
+    ```go
+    //go:build tools
+    // +build tools
+
+    package my_tools
+
+    import (
+        _ "github.com/the/tool"
+        _ "golang.org/x/tools/cmd/stringer"
+    )
+    ```
+
+2. Run `bazel run @rules_go//go mod tidy` to populate the `go.mod` file with the dependencies of the tools.
+
+Instead, if you want the tools' dependencies to be resolved independently of the dependencies of your regular code (experimental):
+
+2. Run `bazel run @rules_go//go mod init` in the directory containing the `tools.go` file to create a new `go.mod` file and then run `bazel run @rules_go//go mod tidy` in that directory.
+3. Add `common --experimental_isolated_extension_usages` to your `.bazelrc` file to enable isolated usages of extensions.
```

Comment:
Nit: reference link on this flag as well as related design consideration would give our users more context to make the appropriate choice.

Reply:
Added in https://github.com/bazelbuild/rules_go/pull/3800